src/test/java/org/elasticsearch/test/ElasticsearchTestCase.java

```diff
@@ -168,7 +168,20 @@ public static boolean awaitBusy(Predicate<?> breakPredicate) throws InterruptedE
     }
 
     public static boolean awaitBusy(Predicate<?> breakPredicate, long maxWaitTime, TimeUnit unit) throws InterruptedException {
+        return awaitBusy(breakPredicate, maxWaitTime, 0, unit);
+    }
+
+    /**
+     * Repeat a task until it returns true or after a given wait time.
+     * By default, it will increase by 2 the sleep time between two runs
+     * @param breakPredicate test you want to run
+     * @param maxWaitTime maximum time you want to wait
+     * @param maxSleepTime if >0 maximum sleep time before two runs
+     * @param unit time unit used for maxWaitTime and maxSleepTime
+     */
+    public static boolean awaitBusy(Predicate<?> breakPredicate, long maxWaitTime, long maxSleepTime, TimeUnit unit) throws InterruptedException {
         long maxTimeInMillis = TimeUnit.MILLISECONDS.convert(maxWaitTime, unit);
+        long maxSleepTimeInMillis = TimeUnit.MILLISECONDS.convert(maxSleepTime, unit);
         long iterations = Math.max(Math.round(Math.log10(maxTimeInMillis) / Math.log10(2)), 1);
```

Comment:
We could... but I think it might be easier to read to just use `sum` as a condition to terminate the loop instead of `iterations`. For instance (untested):

```java
    // After 1s, we stop growing the sleep interval exponentially and just sleep 1s until maxWaitTime
    private static final long AWAIT_BUSY_THRESHOLD = 1000L;

    public static boolean awaitBusy(Predicate<?> breakPredicate, long maxWaitTime, TimeUnit unit) throws InterruptedException {
        long maxTimeInMillis = TimeUnit.MILLISECONDS.convert(maxWaitTime, unit);
        long timeInMillis = 1;
        long sum = 0;
        while (sum + timeInMillis < maxTimeInMillis) {
            if (breakPredicate.apply(null)) {
                return true;
            }
            Thread.sleep(timeInMillis);
            sum += timeInMillis;
            timeInMillis = Math.min(AWAIT_BUSY_THRESHOLD, timeInMillis * 2);
        }
        timeInMillis = maxTimeInMillis - sum;
        Thread.sleep(timeInMillis);
        return breakPredicate.apply(null);
    }
```